src/router.js

```diff
@@ -70,16 +57,45 @@ export class Router {
     return this;
   }
 
+  createRootedPath(fragment) {
+    let path = '';
+
+    if (this.baseUrl.length && this.baseUrl[0] !== '/') {
+      path += '/';
+    }
+
+    path += this.baseUrl;
+
+    if (path[path.length - 1] != '/' && fragment[0] != '/') {
+      path += '/';
+    }
+
+    return normalizeAbsolutePath(path + fragment, this.history._hasPushState);
+  }
+
   navigate(fragment, options) {
-    if(!this.isConfigured && this.parent){
+    if (!this.isConfigured && this.parent) {
       return this.parent.navigate(fragment, options);
     }
 
-    fragment = join(this.baseUrl, fragment);
-    if(fragment === '') fragment = '/';
+    if (fragment === '') {
+      fragment = '/';
+    }
+
+    if (isRooted.test(fragment)) {
+      fragment = normalizeAbsolutePath(fragment, this.history._hasPushState);
+    } else {
+      fragment = this.createRootedPath(fragment);
+    }
+
     return this.history.navigate(fragment, options);
   }
 
+  navigateToRoute(route, params, options) {
```

Comment:
I think navigating with a route + params will be much more common than navigating with a fragment. Should we consider swapping the names, so that the shorter `navigate()` handle the more common scenario? Or maybe we can come up with a shorter name for this... `go()` is nice and short, but it would make the two methods pretty ambiguous.